### DISCUSSION TITLE
Multiline text in mark_text?

### DISCUSSION THREAD

**Discussion by arogozhnikov** [vega/altair#2674] [Q&A] [answered]
There is [a related issue](https://github.com/altair-viz/altair/issues/262) for axes labels and titles that suggests passing lists of string for multilines, but it does not work for just pandas columns with multiline strings that are passed to mark_text. 

If more context is needed, I want to display multiline text over the heatmap, like in [this](https://altair-viz.github.io/gallery/layered_heatmap_text.html) example.

**Comment by joelostblom** [comment, marked as answer]
`mark_text` works the same as axes labels and titles. Pandas columns that contain lists will be split onto multiple lines (works for both string and numbers):

```py
import pandas as pd
import altair as alt


alt.Chart(pd.DataFrame({'a': [['one', 'two']]})).mark_text().encode(
    x=alt.value(0),
    y=alt.value(0),
    text='a'
)
```

![image](https://user-images.githubusercontent.com/4560057/187218970-b459cd98-b26a-47a9-8a3d-1212f570ca72.png)

**Reply by arogozhnikov** [reply]
thank you for your example. The same approach did not work for me, and after digging deeper I found that following example works differently:

- in jupyter+altair it plots multilines
- in jupyter+streamlit (same python env) it plots stringified version of array in a single line

I believe the reason is that streamlit uses old vega-lite maybe? I can confirm after using .to_html() in streamlit the porblem is gone

```python
import pandas as pd
import altair as alt

def test_plot():
    df = pd.DataFrame([
        {'text': f'this is {row}', 'i': row // 10, 'j': row % 10, 'value': row}
        for row in range(100)
    ])
    df['text_split'] = df['text'].str.split(' ')

    base = alt.Chart(df).encode(
        x='i:N',
        y='j:N',
        tooltip=list(df.columns)
    )

    return alt.layer(
        base.mark_rect().encode(
            color='value:Q',
        ),
        base.mark_text(baseline='middle').encode(
            text='text_split'
        ),
    ).properties(
        width=500,
        height=500,
    )
```
